app/main.py

```diff
@@ -2,0 +1,16 @@
+from app.cafe import Cafe
+from app.errors import NotVaccinatedError, NotWearingMaskError
+
+
+def go_to_cafe(friends: list[dict], cafe: Cafe) -> str:
+    masks_to_buy = 0
+    for visitor in friends:
+        try:
+            cafe.visit_cafe(visitor)
+        except NotWearingMaskError:
+            masks_to_buy += 1
+        except NotVaccinatedError:
+            return "All friends should be vaccinated"
+    if masks_to_buy != 0:
+        return f"Friends should buy {masks_to_buy} masks"
+    return f"Friends can go to {cafe.name}"
```

Comment:
The logic should ensure that 'Friends can go to {cafe.name}' is only returned if all friends are allowed to visit. Currently, it returns this message after the first successful visit, which may not be correct if subsequent friends have issues.